### PR TITLE
TSCH: re-enable TSCH regression tests

### PR DESCRIPTION
**Pull request by yatch**
The main change is to use `cooja` mote instead of `z1`.

Two other changes:
- `20-rpl-tsch-orchestra.csc` failed because of a link error for `nbr_routes`. This is resolved by adding a `#if` guard in `orchestra-rule-unicast-per-neighbor-rpl-storing.c`. There could be a better solution.
- The test criteria seem incorrect. They are updated by the following.
```diff
-WAIT_UNTIL(msg.endsWith("Routing links (8 in total):"));
+WAIT_UNTIL(msg.endsWith("Routing links (9 in total):"));
```

### CODE REVIEW

**Comment by yatch** [comment]
Looks good. https://travis-ci.org/contiki-os/contiki/jobs/193198432

```shell
$ make -C regression-tests/??-$BUILD_TYPE RUNALL=true summary
make: Entering directory `/home/travis/build/contiki-os/contiki/regression-tests/11-ipv6'
Running test 01-cooja-ipv6-udp with random Seed 1: ................ OK
Running test 04-exp5438-udp-fragmentation-contikimac with random Seed 1: ..................... OK
Running test 05-exp5438-udp-fragmentation-nullrdc with random Seed 1: .................. OK
Running test 06-exp5438-udp-fragmentation-contikimac-large-sender with random Seed 1: .................... OK
Running test 07-exp5438-udp-fragmentation-nullrdc-large-sender with random Seed 1: .................. OK
Running test 08-exp5438-udp-fragmentation-contikimac-too-large-sender with random Seed 1: .................. OK
Running test 09-exp5438-udp-fragmentation-nullrdc-too-large-sender with random Seed 1: .................. OK
Running test 11-exp5438-unicast-fragmentation-nullrdc with random Seed 1: .................... OK
Running test 12-exp5438-unicast-fragmentation-contikimac-large-sender with random Seed 1: ................. OK
Running test 13-exp5438-unicast-fragmentation-nullrdc-large-sender with random Seed 1: ................. OK
Running test 14-exp5438-unicast-fragmentation-contikimac-too-large-sender with random Seed 1: ................. OK
Running test 15-exp5438-unicast-fragmentation-nullrdc-too-large-sender with random Seed 1: ..................... OK
Running test 16-sky-ipv6-rpl-collect with random Seed 1: ........................... OK
Running test 17-cooja-multicast-11-hops with random Seed 1: ............ OK
Running test 18-cooja-multicast-31-hops with random Seed 1: ........ OK
Running test 19-rpl-tsch with random Seed 1: ............... OK
Running test 20-rpl-tsch-orchestra with random Seed 1: ................. OK
Running test 21-rpl-tsch-security with random Seed 1: ................. OK
make: Leaving directory `/home/travis/build/contiki-os/contiki/regression-tests/11-ipv6'
```

File: apps/orchestra/orchestra-rule-unicast-per-neighbor-rpl-storing.c [hunk 1]
```diff
@@ -45,6 +45,10 @@
 #include "net/ipv6/uip-ds6-route.h"
 #include "net/packetbuf.h"
 #include "net/rpl/rpl-conf.h"
+#include "net/rpl/rpl-private.h"
+
+#if (RPL_MOP_DEFAULT == RPL_MOP_STORING_NO_MULTICAST) || \
+  (RPL_MOP_DEFAULT == RPL_MOP_STORING_MULTICAST)
```
Comment (simonduq) [inline apps/orchestra/orchestra-rule-unicast-per-neighbor-rpl-storing.c]: for generality I'd rather check if `UIP_MAX_ROUTES` is 0 or not (when 0, `nbr_routes` is not compiled)

**Comment by simonduq** [comment]
Wonderful!! Very helpful :)
I wonder why you had to increase the link count from 8 to 9. Would you have the mote output with list of links at hand? (on the go and unable to run it now..)

File: apps/orchestra/orchestra-rule-unicast-per-neighbor-rpl-storing.c (hunk 1 above)
Comment (yatch) [inline apps/orchestra/orchestra-rule-unicast-per-neighbor-rpl-storing.c]: That's nice. I'll borrow your idea.

**Comment by yatch** [comment]
@simonduq Here is the output by `print_network_status()` in `node.c` when the test succeeds:
```shell
02:00.382       ID:1    --- Network status ---
02:00.382       ID:1    - Server IPv6 addresses:
02:00.382       ID:1    -- fd00::201:1:1:1
02:00.382       ID:1    -- fe80::201:1:1:1
02:00.382       ID:1    - Default route:
02:00.382       ID:1    -- None
02:00.382       ID:1    - Routing links (9 in total):
02:00.382       ID:1    -- fd00::202:2:2:2 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::206:6:6:6 to fd00::202:2:2:2 (lifetime: 1748 seconds)
02:00.382       ID:1    -- fd00::203:3:3:3 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::204:4:4:4 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::208:8:8:8 to fd00::204:4:4:4 (lifetime: 1791 seconds)
02:00.382       ID:1    -- fd00::207:7:7:7 to fd00::208:8:8:8 (lifetime: 1764 seconds)
02:00.382       ID:1    -- fd00::209:9:9:9 to fd00::208:8:8:8 (lifetime: 1770 seconds)
02:00.382       ID:1    -- fd00::205:5:5:5 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    ----------------------
```
The number of the printed entries is 8, however there is another which is for the root node.

I'd say this was a bug. Sorry for my laziness. I should have made a separate commit to change the test criteria. Besides, it'd be nice to print an entry which doesn't have its parent, that is, the root node in order to keep consistency between a return value by `rpl_ns_num_nodes()` and the printed list.

This is the output by a new version of `print_network_status()`, which will be pushed later. 
```shell
02:00.382       ID:1    --- Network status ---
02:00.382       ID:1    - Server IPv6 addresses:
02:00.382       ID:1    -- fd00::201:1:1:1
02:00.382       ID:1    -- fe80::201:1:1:1
02:00.382       ID:1    - Default route:
02:00.382       ID:1    -- None
02:00.382       ID:1    - Routing links (9 in total):
02:00.382       ID:1    -- fd00::201:1:1:1 --- DODAG root  (lifetime: 4294967295 seconds)
02:00.382       ID:1    -- fd00::202:2:2:2 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::206:6:6:6 to fd00::202:2:2:2 (lifetime: 1748 seconds)
02:00.382       ID:1    -- fd00::203:3:3:3 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::204:4:4:4 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    -- fd00::208:8:8:8 to fd00::204:4:4:4 (lifetime: 1791 seconds)
02:00.382       ID:1    -- fd00::207:7:7:7 to fd00::208:8:8:8 (lifetime: 1764 seconds)
02:00.382       ID:1    -- fd00::209:9:9:9 to fd00::208:8:8:8 (lifetime: 1770 seconds)
02:00.382       ID:1    -- fd00::205:5:5:5 to fd00::201:1:1:1 (lifetime: 1779 seconds)
02:00.382       ID:1    ----------------------
```

**Comment by yatch** [comment]
Updates the PR:
- rename .csc files including `-cooja-` in their names to express which platform is used
- make a separate commit to change the test criteria
- make `print_network_status()` in `node.c` to print an entry for the root

**Comment by simonduq** [comment]
Awesome, thanks! 👍

**Pull request merged**